### PR TITLE
Add callback_url option

### PR DESCRIPTION
**Pull request by janxious**
More control over the async lifecycle for CLI users and testing. Hand tested with ngrok + nc.

### CODE REVIEW

File: bin/docraptor [hunk 1]
```diff
@@ -53,6 +53,14 @@ opt_parser = OptionParser.new do |opts|
     options[:async] = async
   end
 
+  opts.on("--callback_url url", "Set a callback URL to be notified upon success. Defaults to naught") do |callback_url|
```
Comment (jqr) [inline bin/docraptor]: I think "defaults to naught" is confusing, would it be alright to just remove it?

File: bin/docraptor [hunk 2]
```diff
@@ -53,6 +53,14 @@ opt_parser = OptionParser.new do |opts|
...
+    # callback_url can be used for local testing. You need three consoles:
+    #   nc -lk 9001                                                       # listen on port 9001 for any data
+    #   ngrok http 9001                                                   # setup external URL; port from nc session
+    #   bin/docraptor content --callback_url "http://1111111111.ngrok.io" # copied url from the ngrok session
```
Comment (jqr) [inline bin/docraptor]: How about putting this into an example or the help text? I'm concerned users won't look here for instructions.

**Comment by jqr** [comment]
👍 other than minor comments.

File: bin/docraptor (hunk 2 above)
Comment (janxious) [inline bin/docraptor]: Do we have a place for documenting this tool? It seems like too much information for the command line output from `--help`/`--usage`

Comment (jqr) [inline bin/docraptor]: We have a README and we could add an `examples/` dir.

Comment (jqr) [inline bin/docraptor]: There's probably some relatively easy way to do long form help within the `--help` output, but this seems more like manpage material.

Comment (janxious) [inline bin/docraptor]: I added a big section to the README about testing callbacks, see whatcha think

Comment (jqr) [inline bin/docraptor]: It just occurred to me we also have the wiki available for this sort of stuff.

Comment (janxious) [inline bin/docraptor]: Yar. It's not the worst idea to document all the options there, but I worry about synchronization issues. Could probably document our example cases there?

Comment (jqr) [inline bin/docraptor]: Mostly I think `examples/` is where we put these in the other client repos, but any of the solutions can work.

Comment (janxious) [inline bin/docraptor]: In this case for `examples`, do you think executable `.sh` files?

Comment (jqr) [inline bin/docraptor]: Sure! Makes sense to me. Examples don't necessarily need to be executable on their own, but it's really nice if they are.

**Pull request merged**